src/Nullable.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { EqualityComparer } from "./EqualityComparer"
+
+/**
+ * The `Nullable` type is, unsurprisingly, the opposite of the built-in `NonNullable`
+ * type. Sometimes, it is advantageous to model values as `Nullable` instead of using
+ * the `Option` type. This module is designed to provide a set of useful functions for
+ * seamlessly working with `Nullable` values in function pipelines.
+ *
+ * _Some_ of this behavior can be reproduced with JavaScript's nullish-coalescing `??`
+ * operator, and the nullish-safe accessor `?.` operator. However, those operators lend
+ * themselves primarily to OO-style programming, not function pipelining or coposition.
+ * These functions are curried and are designed to be used with right-to-left function
+ * composition like `pipe` and `flow`.
+ *
+ * @example
+ * declare const str: Nullable<string>
+ *
+ * pipe(
+ *     str,
+ *     Nullable.map(s => `¡${s}!`),
+ *     Nullable.map(String.reverse),
+ *     Nullable.defaultWith(() => "")
```

Comment:
:pick: Should we use `defaultValue` here instead of `defaultWith`?

Reply:
https://github.com/taylorn16/fp-toolkit/commit/486cdbf07f3e4d23c6335dbf976713dbd5baf8a7